### PR TITLE
Add browser compatibility

### PR DESCRIPTION
**Pull request by wischli**
## https://github.com/KILTprotocol/ticket/issues/345
Unfortunately, the API did not work in a browser out of the box. One issue was, that file storage (e.g. `fs.readFileSync` for WASM buffer in `wasm_exec.js`) is not available in the browser and vise-versa `fetch` not in node by default. Since `require` is also missing in the browser.

I only had to change some bits of code and added a full working example 3aa60ab and a README 99f2c61 (updated in e1cac05).

## How to test:
Obviously, step 1 only needs to be done now. After this PR has been merged and published to NPM, it is not necessary anymore.

0. Clone repo and build (in case you haven't already)
```
mkdir -p $GOPATH/src/github.com/KILTprotocol/
git clone https://github.com/KILTprotocol/portablegabi.git \
  $GOPATH/src/github.com/KILTprotocol/portablegabi
cd $GOPATH/src/github.com/KILTprotocol/portablegabi
git checkout wf-add-browser-compatibility
yarn build
```
1. Link with yarn to be able to use browser compatible `wasm_exec.js` and retrieve `main.wasm` (which is located in build)
```
pushd build && yarn link && popd
```
2. Install Portablegabi and Webpack to browser example
```
cd docs/examples/browser
yarn install
yarn link "@kiltprotocol/portablegabi"
yarn webpack
```
3. Serve to a server of your choice inside `docs/examples/browser`, e.g. goexec:
```
goexec 'http.ListenAndServe(":8080", http.FileServer(http.Dir(".")))'
# OR
python -m http.server 8080
```
4. Navigate to http://localhost:8080 and open console
5. Should not see errors:
Screenshot 2020-04-17 at 09 10 13

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](https://deviq.com/boy-scout-rule/)
- [x] I have documented the changes (where applicable)

### CODE REVIEW

**Comment by wischli** [comment]
I would like @tjwelde to review this as well.

**Comment by wischli** [comment]
- [ ] Figure out how to load WASM without relative paths (check out Polkadot)

File: docs/examples/browser/README.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+Most certainly, you can the Portablegabi API in the browser.
```
Comment (tjwelde) [inline docs/examples/browser/README.md:1]: missing word

File: docs/examples/browser/README.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+But you have to make it compatible with the browser API first, e.g. by bundling it with webpack.
+
+Please note that **bundling your JS code does not suffice unfortunately**.
+You also need to **serve the Portablegabi WASM file** [`main.wasm`] located in `/build/src/wasm`.
+We recommend using our webpack configuration below which copies the WASM to your `dist` directory, in which your `bundle.js` should be stored.
+Please **check out the FAQ** in case your `node_modules` directory is not in the root level of your app, or you have a custom webpack output folder apart from `dist`.
+
+## General Setup
+
+1.  Bundle your JavaScript or TypeScript file that runs all the, e.g. with `yarn webpack --entry <your_script_file> --out <desired_output_file>
```
Comment (tjwelde) [inline docs/examples/browser/README.md:11]: `that runs all the` what? feels like something is missing here

File: docs/examples/browser/README.md (hunk 1 above)
Comment (wischli) [reply docs/examples/browser/README.md]: Resolved in 1927391

File: docs/examples/browser/README.md (hunk 2 above)
Comment (wischli) [reply docs/examples/browser/README.md]: Resolved in 1927391

**Comment by wischli** [comment]
> You might want to exclude all the docs when publishing to npm. But this can be done in another ticket.

Only the `build` directory is published to npm, so the docs are already excluded. You might have gotten confused because of locally linking the portablegabi module here to check the browser compatibilty, which is not unnecessary after merging this PR.

Screenshot 2020-05-12 at 11 57 29

**Pull request merged**